shpc/main/modules/base.py

```diff
@@ -331,152 +336,94 @@ def check(self, module_name):
+        if not os.path.exists(module.module_dir):
             logger.exit(
-                "%s does not exist. Is this a known registry entry?" % module_dir
+                "%s does not exist. Is this a known registry entry?" % module.module_dir
             )
 
-        config = self._load_container(module_name.rsplit(":", 1)[0])
-        return self.container.check(module_name, config)
+        return module.check()
 
-    def install(
-        self, name, tag=None, view=None, disable_view=False, force=False, **kwargs
-    ):
+    def new_module(self, name, tag=None, tag_exists=True):
         """
-        Given a unique resource identifier, install a recipe.
-
-        For lmod, this means creating a subfolder in modules, pulling the
-        container to it, and writing a module file there. We've already
-        grabbed the name from docker (which is currently the only supported).
+        Create a new module
         """
         name = self.add_namespace(name)
 
         # If the module has a version, overrides provided tag
         if ":" in name:
             name, tag = name.split(":", 1)
-        config = self._load_container(name, tag)
-
-        # The chosen tag is set for the config (or defaults to latest)
-        if not config.tag:
-            logger.exit(
-                "%s is not a known identifier. Choices are:\n%s"
-                % (name, "\n".join(config.tags.keys()))
-            )
 
-        # We currently support gh, docker, path, or oras
-        uri = config.get_uri()
+        module = Module(name)
+        module.config = self._load_container(module.name, tag)
 
-        # If we have a path, the URI comes from the name
-        if ".sif" in uri:
-            uri = name.split(":", 1)[0]
+        # Ensure the tag exists, if required, uses config.tag
+        if tag_exists:
+            module.validate_tag_exists()
 
-        # This is a tag object with name and digest
-        tag = config.tag
+        # Pass on container and settings
+        module.container = self.container
+        module.settings = self.settings
+        return module
 
-        # Pull the container to the module directory OR container base
-        module_dir = os.path.join(self.settings.module_base, uri, tag.name)
-        subfolder = os.path.join(uri, tag.name)
-        container_dir = self.container.container_dir(subfolder)
-
-        # Are we also installing to a named view?
-        if view is None and not disable_view:
-            view = self.settings.default_view
-
-        # We only want to load over-rides for a tag at install time
-        config.load_override_file(tag.name)
-
-        # A view is a symlink under views_base/$view/$module
-        if view:
-            if view not in self.views:
-                logger.exit(
-                    "View %s does not exist, shpc view create %s." % (view, view)
-                )
+    def install(self, name, tag=None, force=False, **kwargs):
```

Comment:
This `force` is not used anymore !

Reply:
Huh -  I wonder why we had it here before? When you refactor upgrade it likely will make sense to have meaning here - e.g., an install with `--force` would remove stuff and re-install (if I remember you are working on this in the other PR).

Reply:
Before, it was there for overwriting symlinks in views, which is now in `view_install`. Yes, it may gain a new meaning (and a new name ?) in the other PR, so I wouldn't remove it just yet